lsp-terraform.el

```diff
@@ -0,0 +1,62 @@
+;;; lsp-terraform.el --- Terraform Client settings         -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2019 Ross Donaldson
+
+;; Author: Ross Donaldson
+;; Keywords: terraform lsp
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; LSP client for Terraform
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(defgroup lsp-terraform nil
+  "LSP support for Terraform, using terraform-lsp"
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/juliosueiras/terraform-lsp")
+  :package-version `(lsp-mode . "6.2"))
+
+(defcustom lsp-terraform-server "terraform-lsp"
+  "Path to the `terraform-lsp' binary."
+  :group 'lsp-terraform
+  :risky t
+  :type 'file
+  :package-version `(lsp-mode . "6.2"))
+
+(defcustom lsp-terraform-enable-logging nil
+  "If non-nil, enable `terraform-ls''s native logging."
+  :group 'lsp-terraform
+  :risky t
+  :type 'boolean
+  :package-version `(lsp-mode . "6.2"))
+
+(defun lsp-terraform--make-launch-cmd ()
+  (-let [base `(,lsp-terraform-server)]
+    (if lsp-terraform-enable-logging
```

Comment:
use `when` when there is one leg.

Reply:
Ah, happy to. Wanted to leave a clear place for more configurations in the future, but this total makes sense to me for now :+1: